File: Easy/680. Valid Palidrome 2/Python/Solution.py
def validPalindrome(self, s: str) -> bool:
    left = 0
    right = len(s) - 1
    s_list = list(s)
    flag = True
    while left <= right:
        if s_list[left] != s_list[right] and flag:
            new_right = right - 1
            new_left = left
            while new_left <= new_right:
                if  s_list[new_right] != s_list[new_left]:
                    flag = False
                new_left += 1
                new_right -= 1

            if flag:
                return True

            new_right = right
            new_left = left + 1
            flag = True
            while new_left <= new_right:
                if  s_list[new_right] != s_list[new_left]:
                    flag = False
                new_left += 1
                new_right -= 1

            return flag

        left += 1
        right -= 1

    return True

File: Easy/680. Valid Palidrome 2/C++/Solution.cpp
// 680. Valid Palindrome II
// Given a string s, return true if the s can be palindrome after deleting at most one character from it.
// A palindrome is a string that reads the same backward as forward.
// Example 1:
// Input: s = "aba"
// Output: true
// Example 2:
// Input: s = "abca"
// Output: true
// Example 3:
// Input: s = "abc"
// Output: false
// Constraints:
// 1 <= s.length <= 105
// s consists of lowercase English letters.
// approach: Two pointers

// Solution:
#include <iostream>
#include <string>
using namespace std;


bool validPalindrome(string s) {
    int left = 0; 
    int right = s.size() - 1;
    bool flag = true;
    while (left <= right){
        if (s[left] != s[right])
        {
            int new_left = left;
            int new_right = right - 1;
            while (new_left <= new_right)
            {
                if (s[new_left] != s[new_right])
                {
                    flag = false;
                    break;
                }
                new_left++;
                new_right--;
            }

            if (flag)
            {
                return true;
            }
            flag = true;
            new_left = left + 1;
            new_right = right;

            while (new_left <= new_right)
            {
                if (s[new_left] != s[new_right])
                {
                    flag = false;
                    break;
                }
                new_left++;
                new_right--;
            }

            return flag;

        }
        left++;
        right--;
    }
    return flag;
}